### PR TITLE
New package: notepadqq

### PR DESCRIPTION
**Pull request by cr6git**
Template based on the Arch [PKGBUILD](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/notepadqq).

### CODE REVIEW

File: srcpkgs/notepadqq/template [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+do_fetch() {
+	mkdir -p ${wrksrc} && cd ${wrksrc}
+	printf "\033[1;35m=> Fetching source from ${_notepadqq} ...\033[m\n"
+	git clone ${_notepadqq}
+	cd notepadqq && git reset --hard ${_release}
+	cd ..
+	printf "\033[1;35m=> Fetching source from ${_CodeMirror} ...\033[m\n"
+	git clone ${_CodeMirror}
+}
```
Comment (Gottox) [inline srcpkgs/notepadqq/template]: Why not using the provided tarballs?

Comment (cr6git) [inline srcpkgs/notepadqq/template]: Using the tarballs the git commands failed: 

pre_configure() {
	git config submodule.src/editor/libs/codemirror.url ../CodeMirror
	git submodule update --init
}

Comment (ebfe) [inline srcpkgs/notepadqq/template]: You can fetch the corresponding codemirror tarball too and extract it manually

Comment (cr6git) [inline srcpkgs/notepadqq/template]: Using the tarballs it fails: 
````
=> notepadqq-1.0.1_1: running pre_configure ...
/builddir/notepadqq-1.0.1/notepadqq-1.0.1
fatal: not in a git directory
````
Do you have a better idea ?

Comment (ebfe) [inline srcpkgs/notepadqq/template]: Don't run git submodule stuff, fetch the corresponding codemirror tarball and extract it.

Comment (cr6git) [inline srcpkgs/notepadqq/template]: Without git submodule stuff I get new errors: see [the log](https://bpaste.net/show/22af9bb9be34) ...

Comment (ebfe) [inline srcpkgs/notepadqq/template]: The submodule lives at `src/editor/libs/codemirror` so you need to extract it there.
See https://github.com/notepadqq/notepadqq/blob/master/.gitmodules

Comment (cr6git) [inline srcpkgs/notepadqq/template]: OK. I created this [new template](https://bpaste.net/show/56ea28cb10c8), moving CodeMirror to the right folder, but I get  new errors: see my [new log](https://bpaste.net/show/3ddc967a7683) ...

Comment (Gottox) [inline srcpkgs/notepadqq/template]: Don't branch for hostmakedepends when on crossbuild! It's only valid in a few cornercases!

Comment (ebfe) [inline srcpkgs/notepadqq/template]: @cr6git just use the revision referenced by the submodule 
`_Code_release=d790fc39c1a5f06aa66415110b8ebe3026df665a`

Comment (cr6git) [inline srcpkgs/notepadqq/template]: @ebfe Exact!! this commit solves the problem, now I can build the package for my current arch x86_64, using [this template](https://bpaste.net/show/812beb762081). 
@Gottox I can't find another solution regarding cross-compilation.

Comment (cr6git) [inline srcpkgs/notepadqq/template]: What should I do ? new git push with these changes ?

Comment (Gottox) [inline srcpkgs/notepadqq/template]: just remove the branch around hostmakedepends.

Comment (cr6git) [inline srcpkgs/notepadqq/template]: Done.

**Comment by cr6git** [comment]
@Gottox I hope the changes are what you want.

**Comment by cr6git** [comment]
@Gottox I removed `qt5-tools-devel` from this template, I think it's better.

**Comment by cr6git** [comment]
I think this new template is cleaner.

**Comment by cr6git** [comment]
Is there a reason why this PR is still blocked ?

**Pull request merged**